File: api/DTOs/Loan/LoanDto.cs
namespace api.DTOs.Loan
    {
    public class LoanDto
        {
        public int Id { get; set; }
        public int BookId { get; set; }
        public string BookTitle { get; set; }
        public string UserId { get; set; }
        public string UserName { get; set; }
        public DateTime LoanDate { get; set; }
        public DateTime DueDate { get; set; }
        public DateTime? ReturnDate { get; set; }
        public string Status { get; set; }
        public int DaysRemaining { get; set; }
        public bool IsOverdue { get; set; }
        }
    }

File: client/src/app/shared/dtos/loan.dto.ts
export interface LoanDto {
  id: number;
  bookId: number;
  bookTitle: string;
  userId: string;
  userName: string;
  loanDate: string;
  dueDate: string;
  returnDate: string | null;
  status: string; // 'Active' | 'Returned' | 'Overdue'
  daysRemaining: number;
  isOverdue: boolean;
}

export interface CreateLoanDto {
  bookId: number;
  userId: string;
  customDueDate?: string;
}

export interface LoanFilterDto {
  userId?: string;
  bookId?: number;
  status?: string;
  isOverdue?: boolean;
  pageNumber: number;
  pageSize: number;
}
